### PR TITLE
Improve Cadence 1.0 migration

### PR DESCRIPTION
**Pull request by turbolent**
Work towards onflow/cadence#3096

- Add support for localnet
- Update Cadence, to get a fix for the migration in: https://github.com/onflow/cadence/pull/3103
- Fix building static binary (glibc incompatibility issue)
- Fix burner contract deployment and imports in updated system contracts. 
  Caused by https://github.com/onflow/flow-core-contracts/commit/7204ba8f720f3b8186a7ffb087f971919b42dd43#diff-1803784ae7aa05d34fc981d144045dba79bac0d9a866c1b0d8df69ef3e8bea85L105-R110
- Optimize `CadenceBaseMigrator.InitMigration`: Instead of building a full payload snapshot, which is only used to get contract code for any account, only gather all contract code registers and build a mapping for it
- Cache program loading errors, as subsequent loads will never result in a success
- Only print program loading errors once

### CODE REVIEW

**Review by SupunS — APPROVED** [review]

**Review by SupunS — APPROVED** [review]

File: cmd/util/ledger/migrations/change_contract_code_migration.go [hunk 1]
```diff
@@ -292,64 +324,64 @@ func SystemContractChanges(chainID flow.ChainID, options SystemContractChangesOp
...
 		NewSystemContractChange(
 			systemContracts.FlowToken,
 			coreContracts.FlowToken(
-				systemContracts.FungibleToken.Address.HexWithPrefix(),
+				fungibleTokenAddress.HexWithPrefix(),
 				fungibleTokenMetadataViewsAddress.HexWithPrefix(),
-				systemContracts.MetadataViews.Address.HexWithPrefix(),
-				systemContracts.ViewResolver.Address.HexWithPrefix(),
+				metadataViewsAddress.HexWithPrefix(),
+				burnerAddress.HexWithPrefix(),
```
Comment (turbolent) [inline cmd/util/ledger/migrations/change_contract_code_migration.go:350]: flow-ft's contract code "constructor function" changed in a recent commit: the argument is no longer the address of the view resolver, but the one of the burner

**Comment by turbolent** [comment]
More improvements are needed, but that should not block getting these improvements in

File: cmd/util/ledger/migrations/cadence_values_migration.go [hunk 2]
```diff
@@ -220,23 +252,59 @@ func NewCadence1CapabilityValueMigrator(
...
+
+	// Only report program loading errors once,
+	// omit full error message for subsequent occurrences
+
+	var programLoadingError environment.ProgramLoadingError
+	if errors.As(err, &programLoadingError) {
+		location := programLoadingError.Location
+		_, ok := t.reportedProgramLoadingErrors.LoadOrStore(location, struct{}{})
+		if ok {
+			message = "error getting program"
```
Comment (SupunS) [inline cmd/util/ledger/migrations/cadence_values_migration.go:277]: can just return from here
```suggestion
			return "error getting program"
```

**Review by SupunS — APPROVED** [review]

**Review by janezpodhostnik — APPROVED** [review]

File: cmd/util/ledger/migrations/cadence_values_migration.go (hunk 2 above)
Comment (turbolent) [reply cmd/util/ledger/migrations/cadence_values_migration.go]: Simplified in [388a112](https://github.com/onflow/flow-go/pull/5388/commits/388a11294e702b48d13f62c769a03b4006b0658f)

**Pull request merged**